lib/zinnia.js

```diff
@@ -189,10 +205,15 @@ export async function run ({
       }
     })(),
     (async () => {
-      const [code] = await once(childProcess, 'close')
+      const [code] = await Promise.race(
```

Comment:
I think we should wait for all processes to close before we continue, otherwise we can end up with a module having two processes - the old process still writing the output, the new process already starting.

```suggestion
      const [code] = await Promise.allSettled(
```

Reply:
Good idea. Changed it to `Promise.all()` (don't see the need for `allSettled`, correct me if I'm wrong) and ensured the on exit listener above kills all the procs when one exits